models/xero__general_ledger.sql

```diff
@@ -13,17 +51,20 @@ with journals as (
         j.account_name,
         j.account_type,
         jt.option,
+        jt.tracking_category_id,
         j.description,
         j.gross_amount,
         j.journal_id,
         j.net_amount,
         j.tax_amount,
         j.tax_name,
-        j.tax_type
+        j.tax_type,
+        {{case_when_tracking_category}}
+
     from {{ var('journal_line') }} j
     left join {{  var('journal_line_tracking') }} jt on
         j.journal_line_id = jt.journal_line_id
-    {{ dbt_utils.group_by(14) }}
+    {{ dbt_utils.group_by(15) }}
```

Comment:
Shouldn't this be 14 + the length of the array?

Reply:
Original value was 14, and I added 'jt.tracking_category_id' to the grouping list, that is why it is now 15. 
Inside the `{{case_when_tracking_category}}` there is an aggregating function (like 'sum'): `string_agg(distinct option, ' | ')`. That is why it is not grouped. 
When we are using a column from the array to group, we add '+ the length of the array'